pygeos/predicates.py

```diff
@@ -73,18 +75,77 @@ def is_empty(geometry, **kwargs):
     geometry : Geometry or array_like
         Any geometry type is accepted.
 
+    See also
+    --------
+    is_null : checks if the object is a geometry
+
     Examples
     --------
     >>> is_empty(Geometry("POINT EMPTY"))
     True
     >>> is_empty(Geometry("POINT (0 0)"))
     False
-    >>> is_empty(Empty)
-    True
+    >>> is_empty(NaG)
+    False
     """
     return ufuncs.is_empty(geometry, **kwargs)
 
 
+def is_geometry(geometry, **kwargs):
+    """Returns True if the object is a geometry
+
+    Parameters
+    ----------
+    geometry : Geometry or array_like
+
+    See also
+    --------
+    is_null : the inverse of is_geometry.
+
+    Examples
+    --------
+    >>> is_geometry(Geometry("POINT (0 0)"))
+    True
+    >>> is_geometry(Geometry("GEOMETRYCOLLECTION EMPTY"))
+    True
+    >>> is_geometry(None)
+    False
+    >>> is_geometry(float("nan"))
+    False
+    >>> is_geometry(NaG)
+    False
+    """
+    return ufuncs.is_geometry(geometry, **kwargs)
+
+
+def is_null(geometry, **kwargs):
+    """Returns True if the object is not a geometry (NaG)
+
+    Parameters
+    ----------
+    geometry : Geometry or array_like
+
+    See also
+    --------
+    is_geometry : the inverse of is_null.
+    is_empty : checks if the object is an empty geometry
+
+    Examples
+    --------
+    >>> is_null(Geometry("POINT (0 0)"))
+    False
+    >>> is_null(Geometry("GEOMETRYCOLLECTION EMPTY"))
+    False
+    >>> is_null(None)
+    True
+    >>> is_null(float("nan"))
+    True
```

Comment:
So this is somewhat related to the comment I made below about limiting the allowed input (checking each element from being float gives an overhead, and for me it is not necessary a needed feature to be able to pass all of three potential missing-like objects)